### PR TITLE
[CG] Bump Microsoft.IO.Redist to 6.0.1

### PR DESCRIPTION
**Pull request by steveisok**
CG was flagging ILLink.Tasks and Microsoft.NETCore.Platforms as pulling in an older version of Microsoft.IO.Redist. This change pins the version we use to clear the alert.

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -263,5 +263,7 @@
     <NodePackageVersion>$(runtimewinx64MicrosoftNETCoreRuntimeWasmNodeTransportPackageVersion)</NodePackageVersion>
     <!-- The package path for python in src/mono/mono.proj needs to be updated if this changes-->
     <EmsdkVersion>3.1.56</EmsdkVersion>
+    <!-- Microsoft.IO.Redist version to satisfy CG -->
+    <MicrosoftIORedistVersion>6.0.1</MicrosoftIORedistVersion>
```
Comment (carlossanlop) [inline eng/Versions.props:267]: Does this file change mean this property will be updatable via darc?

**Review by carlossanlop — APPROVED** [review]

**Comment by carlossanlop** [comment]
Do we need to backport this change?

File: eng/Versions.props (hunk 1 above)
Comment (steveisok) [reply eng/Versions.props:267]: Is it available via darc? I didn't think it was. I just put it there to manage the version easier.

Comment (carlossanlop) [reply eng/Versions.props:267]: I don't know either, that's why I asked :)

**Review by sbomer — COMMENTED** [review]
For ILLink.Tasks, this is pulled in as a transitive dependency of Microsoft.Build.Tasks.Core. Could we fix it by bumping MicrosoftBuildVersion to pick up https://github.com/dotnet/msbuild/pull/10375?

**Comment by steveisok** [comment]
> For ILLink.Tasks, this is pulled in as a transitive dependency of Microsoft.Build.Tasks.Core. Could we fix it by bumping MicrosoftBuildVersion to pick up [dotnet/msbuild#10375](https://github.com/dotnet/msbuild/pull/10375)?

Has that shipped yet? I didn't think it did.

**Comment by agocke** [comment]
Let's wait for MSBuild to move first.

**Review by agocke — DISMISSED** [review]
Hold until we decide what to do for MSBuild

**Comment by krwq** [comment]
consider porting this to 8.0/9.0 branches since it also shows on CG alerts there

File: eng/Versions.props (hunk 1 above)
Comment (carlossanlop) [reply eng/Versions.props:267]: I think this should be ok. If darc cannot update it because there are no new versions, then nothing should be affected.

**Comment by ViktorHofer** [comment]
cc @ericstj

**Review by ericstj — CHANGES_REQUESTED** [review]
We shouldn't need this.  We don't reference the `Microsoft.IO.Redist` package at all in main.

That was fixed with the `PackageDownloadAndReference` to only reference MSBuild itself.

@steveisok `main` has no CG alerts for this package.  I only see them in 8.0 and 9.0.  Perhaps we should consider backporting a portion of the audit change from `main` so that we don't end up chasing these alerts for build tasks.

**Comment by agocke** [comment]
> Perhaps we should consider backporting a portion of the audit change from main so that we don't end up chasing these alerts for build tasks.

Which change are you referring to?

**Comment by ericstj** [comment]
This one: https://github.com/dotnet/runtime/pull/107639  I ported it back to 9.0.  We can do something similar for 8.0, though we can't enable audit there.

**Pull request closed without merging**